lnbits/wallets/eclair.py

```diff
@@ -227,3 +227,23 @@ async def paid_invoices_stream(self) -> AsyncGenerator[str, None]:
                     f"lost connection to eclair invoices stream: '{exc}', retrying in 5 seconds"
                 )
                 await asyncio.sleep(5)
+
+    async def sent_payments_stream(self) -> AsyncGenerator[str, None]:
+        while True:
+            try:
+                async with connect(
+                    self.ws_url,
+                    extra_headers=[("Authorization", self.auth["Authorization"])],
+                ) as ws:
+                    while True:
+                        message = await ws.recv()
+                        message = json.loads(message)
+
+                        if message and message["type"] == "payment-sent":
+                            yield message["paymentHash"]
+
+            except Exception as exc:
+                logger.error(
+                    f"lost connection to eclair invoices stream: '{exc}', retrying in 5 seconds"
+                )
+                await asyncio.sleep(5)
```

Comment:
Why do you need a sent payment stream for this extension to work?

Reply:
I need this because when user withdraws from the wallet, hedge position has to be adjusted accordingly to the sats balance in the wallet.

Reply:
Bump @callebtc

Reply:
Ah ok I get the idea. Question: why would you need a payment stream listener for the node backend? Wouldn't it be enough to listen for outgoing payments from an LNbits wallet and adjust the hedge accordingly?

I think the payment stream might be backend-dependent which may imply that this extension can only be used by a subset of backends. Would be nice if it were usable for all of them!

Reply:
>  Wouldn't it be enough to listen for outgoing payments from an LNbits wallet and adjust the hedge accordingly?

I wasn't aware of that. It worked for Eclair backend easily. Could you point on specific LBits events?

Also now it kinda works similarly for incoming and outgoing payments. Maybe it makes sense to keep implementation uniform?